### PR TITLE
feat(projects): Added query option to allow for filtering of user projects

### PR DESCRIPTION
**Pull request by davidenwang**
Modified the OrganizationProjectsEndpoint to support returning just the projects that a user is a member of via team membership. This will eventually be used in the logic of `<NoProjectMessage>` instead of having it look through all of `organization.projects`.

### CODE REVIEW

File: src/sentry/api/endpoints/organization_projects.py [hunk 1]
```diff
@@ -97,6 +97,8 @@ def get(self, request, organization):
                 elif key == "!team":
                     team_list = list(Team.objects.filter(slug__in=value))
                     queryset = queryset.exclude(teams__in=team_list)
+                elif key == "user_projects":
```
Comment (billyvg) [inline src/sentry/api/endpoints/organization_projects.py]: what about like `is_member` or `user_is_member` or `has_membership`?

Comment (davidenwang) [reply src/sentry/api/endpoints/organization_projects.py]: i'll switch over to is_member

File: src/sentry/api/endpoints/organization_projects.py [hunk 2]
```diff
@@ -97,6 +97,8 @@ def get(self, request, organization):
                 elif key == "!team":
                     team_list = list(Team.objects.filter(slug__in=value))
                     queryset = queryset.exclude(teams__in=team_list)
+                elif key == "is_member":
```
Comment (wedamija) [inline src/sentry/api/endpoints/organization_projects.py]: I assume you'll only pass this for non-superusers?

**Review by billyvg — APPROVED** [review]
This looks good to me, idk if @wedamija has additional thoughts

**Review by wedamija — APPROVED** [review]

File: src/sentry/api/endpoints/organization_projects.py (hunk 2 above)
Comment (davidenwang) [reply src/sentry/api/endpoints/organization_projects.py]: yup that's the plan for now

**Pull request merged**